### ISSUE TITLE
Tag[] for actors doesn't save when selected in Properties Window

### ISSUE THREAD

**Issue by KirillianAmu** [FlaxEngine/FlaxEngine#1451]
**Issue description:**
Use of Tag[] doesn't seem to save any tag selected in the Properties Window.
![Desktop_Dimmer_lDVJProPhA](https://github.com/FlaxEngine/FlaxEngine/assets/66353937/8c8e7c47-2e51-45b4-b682-6a9204effb99)

**Steps to reproduce:**

Unpack FlaxBug.zip
Go to Test scene
Select EmptyActor
Try to select the tags via the menu.
Click - expected to save.
Result - Doesn't save.

**Minimal reproduction project:**

[FlaxBug.zip](https://github.com/FlaxEngine/FlaxEngine/files/12667941/FlaxBug.zip)

**Flax version:**
1.6

**Comment by Tryibion** [comment]
I can confirm that this happens for Tag[] member variables. Seems to work fine for Actor.Tags[] and works fine for just Tag.

Was was able to get some sort of error, that seems related.

```
[ 00:00:02.397 ]: [Warning] Inner exception. Could not cast or convert from System.String to FlaxEngine.Tag[].
Stack strace:
   at Newtonsoft.Json.Utilities.ConvertUtils.EnsureTypeAssignable(Object value, Type initialType, Type targetType) in F:\FlaxEngineDotnet\Cache\Intermediate\Deps\NewtonsoftJson\Src\Newtonsoft.Json\Utilities\ConvertUtils.cs:line 641
   at Newtonsoft.Json.Utilities.ConvertUtils.ConvertOrCast(Object initialValue, CultureInfo culture, Type targetType) in F:\FlaxEngineDotnet\Cache\Intermediate\Deps\NewtonsoftJson\Src\Newtonsoft.Json\Utilities\ConvertUtils.cs:line 612
   at Newtonsoft.Json.Serialization.JsonSerializerInternalReader.EnsureType(JsonReader reader, Object value, CultureInfo culture, JsonContract contract, Type targetType) in F:\FlaxEngineDotnet\Cache\Intermediate\Deps\NewtonsoftJson\Src\Newtonsoft.Json\Serialization\JsonSerializerInternalReader.cs:line 1013

[ 00:00:02.397 ]: [Warning] Exception has been thrown during ManagedSerialization::Deserialize. Error converting value "Player" to type 'FlaxEngine.Tag[]'. Path 'TestTags', line 1, position 279.
Stack strace:
   at Newtonsoft.Json.Serialization.JsonSerializerInternalReader.EnsureType(JsonReader reader, Object value, CultureInfo culture, JsonContract contract, Type targetType) in F:\FlaxEngineDotnet\Cache\Intermediate\Deps\NewtonsoftJson\Src\Newtonsoft.Json\Serialization\JsonSerializerInternalReader.cs:line 1017
   at Newtonsoft.Json.Serialization.JsonSerializerInternalReader.CreateValueInternal(JsonReader reader, Type objectType, JsonContract contract, JsonProperty member, JsonContainerContract containerContract, JsonProperty containerMember, Object existingValue) in F:\FlaxEngineDotnet\Cache\Intermediate\Deps\NewtonsoftJson\Src\Newtonsoft.Json\Serialization\JsonSerializerInternalReader.cs:line 335
   at Newtonsoft.Json.Serialization.JsonSerializerInternalReader.SetPropertyValue(JsonProperty property, JsonConverter propertyConverter, JsonContainerContract containerContract, JsonProperty containerProperty, JsonReader reader, Object target) in F:\FlaxEngineDotnet\Cache\Intermediate\Deps\NewtonsoftJson\Src\Newtonsoft.Json\Serialization\JsonSerializerInternalReader.cs:line 1064
   at Newtonsoft.Json.Serialization.JsonSerializerInternalReader.PopulateObject(Object newObject, JsonReader reader, JsonObjectContract contract, JsonProperty member, String id) in F:\FlaxEngineDotnet\Cache\Intermediate\Deps\NewtonsoftJson\Src\Newtonsoft.Json\Serialization\JsonSerializerInternalReader.cs:line 2479
   at Newtonsoft.Json.Serialization.JsonSerializerInternalReader.Populate(JsonReader reader, Object target) in F:\FlaxEngineDotnet\Cache\Intermediate\Deps\NewtonsoftJson\Src\Newtonsoft.Json\Serialization\JsonSerializerInternalReader.cs:line 116
   at Newtonsoft.Json.JsonSerializer.PopulateInternal(JsonReader reader, Object target) in F:\FlaxEngineDotnet\Cache\Intermediate\Deps\NewtonsoftJson\Src\Newtonsoft.Json\JsonSerializer.cs:line 824
   at Newtonsoft.Json.JsonSerializer.Populate(JsonReader reader, Object target) in F:\FlaxEngineDotnet\Cache\Intermediate\Deps\NewtonsoftJson\Src\Newtonsoft.Json\JsonSerializer.cs:line 797
   at FlaxEngine.Json.JsonSerializer.Deserialize(Object input, Byte* jsonBuffer, Int32 jsonLength) in Z:\Flax\Source\FlaxEngine\Source\Engine\Serialization\JsonSerializer.cs:line 328
   at FlaxEngine.Interop.NativeInterop.Invoker.InvokerStaticNoRet3`3.MarshalAndInvoke(Object delegateContext, ManagedHandle instancePtr, IntPtr paramPtr) in Z:\Flax\Source\FlaxEngine\Source\Engine\Engine\NativeInterop.Invoker.cs:line 722
   at FlaxEngine.Interop.NativeInterop.InvokeMethod(ManagedHandle instanceHandle, ManagedHandle methodHandle, IntPtr paramPtr, IntPtr exceptionPtr) in Z:\Flax\Source\FlaxEngine\Source\Engine\Engine\NativeInterop.Unmanaged.cs:line 691
[ 00:00:02.397 ]: [Error] Exception has been thrown during ManagedSerialization::Deserialize.
Error converting value "Player" to type 'FlaxEngine.Tag[]'. Path 'TestTags', line 1, position 279.
```

**Comment by GhettoScouser** [comment]
`[ 00:00:26.777 ]: [Warning] Inner exception. Could not cast or convert from System.String to FlaxEngine.Tag.
Stack strace:
   at Newtonsoft.Json.Utilities.ConvertUtils.EnsureTypeAssignable(Object value, Type initialType, Type targetType)
   at Newtonsoft.Json.Utilities.ConvertUtils.ConvertOrCast(Object initialValue, CultureInfo culture, Type targetType)
   at Newtonsoft.Json.Serialization.JsonSerializerInternalReader.EnsureType(JsonReader reader, Object value, CultureInfo culture, JsonContract contract, Type targetType)

[ 00:00:26.777 ]: [Warning] Exception has been thrown during ManagedSerialization::Deserialize. Error converting value "UITitle" to type 'FlaxEngine.Tag'. Path 'UITag', line 1, position 333.
Stack strace:
   at Newtonsoft.Json.Serialization.JsonSerializerInternalReader.EnsureType(JsonReader reader, Object value, CultureInfo culture, JsonContract contract, Type targetType)
   at Newtonsoft.Json.Serialization.JsonSerializerInternalReader.CreateValueInternal(JsonReader reader, Type objectType, JsonContract contract, JsonProperty member, JsonContainerContract containerContract, JsonProperty containerMember, Object existingValue)
   at Newtonsoft.Json.Serialization.JsonSerializerInternalReader.SetPropertyValue(JsonProperty property, JsonConverter propertyConverter, JsonContainerContract containerContract, JsonProperty containerProperty, JsonReader reader, Object target)
   at Newtonsoft.Json.Serialization.JsonSerializerInternalReader.PopulateObject(Object newObject, JsonReader reader, JsonObjectContract contract, JsonProperty member, String id)
   at Newtonsoft.Json.Serialization.JsonSerializerInternalReader.Populate(JsonReader reader, Object target)
   at Newtonsoft.Json.JsonSerializer.PopulateInternal(JsonReader reader, Object target)
   at Newtonsoft.Json.JsonSerializer.Populate(JsonReader reader, Object target)
   at FlaxEngine.Json.JsonSerializer.Deserialize(Object input, Byte* jsonBuffer, Int32 jsonLength)
   at FlaxEngine.Interop.NativeInterop.Invoker.InvokerStaticNoRet3`3.MarshalAndInvoke(Object delegateContext, ManagedHandle instancePtr, IntPtr paramPtr)
   at FlaxEngine.Interop.NativeInterop.InvokeMethod(ManagedHandle instanceHandle, ManagedHandle methodHandle, IntPtr paramPtr, IntPtr exceptionPtr)
[ 00:00:26.777 ]: [Error] Exception has been thrown during ManagedSerialization::Deserialize.
Error converting value "UITitle" to type 'FlaxEngine.Tag'. Path 'UITag', line 1, position 333.
[ 00:00:26.784 ]: [Info] Loading scene...`

Possibly related?

**Comment by mafiesto4** [comment]
Fixed in 44292efa04c7470cee87bc53abdf64f7c5ca35cb.
